Watchman.Discord/Areas/Initialization/Controllers/InitializationController.cs

```diff
@@ -80,6 +85,65 @@ private async Task MuteRoleInit(Contexts contexts)
             {
                 await _muteRoleInitService.InitForServer(contexts);
             }
+            Log.Information("Mute role initialized");
+        }
+
+        private async Task ReadMessagesHistory(DiscordServerContext server)
+        {
+            Log.Information("Reading messages started");
+            const int LIMIT = 1000;
+
+            foreach (var channel in server.TextChannels)
+            {
+                if (channel.Name.Contains("logs"))
+                    continue;
+
+                var messages = (await _readMessagesHistoryService.ReadMessagesAsync(server, channel, LIMIT)).ToList();
+                if (messages.Count == 0)
+                {
+                    continue;
+                }
+
+                var lastMessageId = messages.Last().messageId;
+                await SaveMessages(messages, channel.Id);
+
+                while (true)
```

Comment:
while(true) to nie jest dobre rozwiązanie 
zawsze jest jakiś sposób na zrobienie tego bezpieczniej

Reply:
może ok opcją będzie podzielenie tego na kilka metod 
albo do{}while zależne od jakiejś zmiennej w środku

Reply:
jeszcze jest kwestia tego czy to powinno być w kontrolerze czy już osobny serwis 
sporo tu obliczeń itp 

poza tym, serwisy mogą przyjmować inne serwisy więc zależności nie są problemem